src/SpriteKit/SKUniform.cs

```diff
@@ -0,0 +1,215 @@
+//
+// SKUniform.cs: SKUniform class
+//
+// Authors:
+//   Vincent Dondain (vidondai@microsoft.com)
+//
+// Copyright 2016 Xamarin Inc.
+//
+
+using System;
+using XamCore.Foundation;
+using XamCore.ObjCRuntime;
+
+using Vector2 = global::OpenTK.Vector2;
+using Vector3 = global::OpenTK.Vector3;
+using Vector4 = global::OpenTK.Vector4;
+using Matrix2 = global::OpenTK.Matrix2;
+using Matrix3 = global::OpenTK.Matrix3;
+using Matrix4 = global::OpenTK.Matrix4;
+
+#if XAMCORE_2_0 || !MONOMAC
+namespace XamCore.SpriteKit {
+	public partial class SKUniform {
+
+		static bool CheckSystemVersion ()
+		{
+#if MONOMAC
+			return PlatformHelper.CheckSystemVersion (10, 12);
+#else
+			return XamCore.UIKit.UIDevice.CurrentDevice.CheckSystemVersion (10, 0);
+#endif // MONOMAC
```

Comment:
The result should be cached (e.g. a `bool?`) as it could be expensive, being used in properties.